test/src/test_main.py

```diff
@@ -0,0 +1,185 @@
+import jsonschema
+import json
+import copy
+import pytest
+
+SCHEMA_PATH = "json-schema/StructuredQuery-schema-v3.json"
+EXAMPLE_JSON_PATH = "example-json/sqv3_all_properties.json"
+
+TERM_CODE = {"code": "somecode-4591", "system": "somesystem-1138", "display": "somedisplay-5832"}
+UNIT = {"code": "somecode-8331", "display": "somedisplay-1269"}
+ATTRIBUTE_CODE = {"code": "somecode-0103", "system": "somesystem-3571", "display": "somedisplay-9567"}
+
+
+@pytest.fixture
+def schema():
+    loaded_schema = None
+    with open(SCHEMA_PATH, 'r') as file:
+        loaded_schema = json.load(file)
+    return loaded_schema
+
+@pytest.fixture
+def basic_sq():
+    sq = {}
+    sq["inclusionCriteria"] = [[ {"termCodes": [TERM_CODE]}]]
+    sq["version"] = "someVersion"
+    return sq
+    
+def test_basic_sq(schema, basic_sq):
+    jsonschema.validate(basic_sq, schema)
+
+def test_example_json(schema):
+    example_json = None
+    with open(EXAMPLE_JSON_PATH, 'r') as file:
+        example_json = json.load(file)
+
+    jsonschema.validate(example_json, schema)
+
+def test_additional_prop_in_criterion(schema, basic_sq):
+    sq = copy.copy(basic_sq)
+    jsonschema.validate(sq, schema)
```

Comment:
Why do you validate the `basic_sq` again every time?

Reply:
I really don't understand what you say. All I see, and I don't have experience in Python, that you validate the structured query after the copy. I assume that directly after the copy, the structured query is identical to the basic structured query.

Reply:
oh i thought you meant all tests in general since you said "every time".  You are right, the tests that dont change the basic sq before adding additional properties dont need to validate the basic sq again. I will remove that.